### PR TITLE
bsc#1210728 - avoid explicid and implicid usage of /tmp filesystem 

### PR DESCRIPTION
**Pull request by angelabriel**
To keep the SAPHanaSR resource agents working even in situations with /tmp filesystem full avoid explicid and implicid usage of /tmp filesystem
Additional added the changes from Peter Pitterling (PR#158 and PR#159) with the changes discussed in the related PRs.
And fixing indentation by exchanging tabs with blanks

### CODE REVIEW

File: ra/SAPHana [hunk 1]
```diff
@@ -1241,50 +1240,49 @@ function check_for_primary() {
...
+                rc=$HANA_STATE_SECONDARY
+                break;;
+            none ) # have seen that mode on second side BEFEORE we registered it as replica
+                rc=$HANA_STATE_STANDALONE
+                break;;
+            * )
+                super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: <$node_status>"
+                dump=$( echo $node_status | hexdump -C );
+                super_ocf_log err "ACT: check_for_primary:  we didn't expect node_status to be: DUMP <$dump>"
+                ((i++))
```
Comment (fmherschel) [inline ra/SAPHana:1277]: Where does this loop counter "i" come from? How is it initialized?
Should we define a local "i" with local i=0 ?

**Review by fmherschel — CHANGES_REQUESTED** [review]
I should be initialized? See my inline comment.

File: ra/SAPHana (hunk 1 above)
Comment (angelabriel) [reply ra/SAPHana:1277]: Commit f44f729 from 20.09.2014 introduced the loop as `for in in 1 2 3 4 5 6 7 8 9` and added the debug message containing the loop counter `i` . With commit eb665c0 from 13.11.2015 the `for i` loop was exchanged with `for chkMethode in gP hU hU hU gP` , but the debug message was not changed even that there was no loop counter `i` anymore. And this missing `i` was not recognized all the last years till Peter's PR.
We can now change the debug message and remove the loop counter. Or we simply add the loop counter inside the loop as the PR proposes. As `i++` initializes the variable there is no need to explicitly initialize the variable. But if it is important to declare the loop count variable used in the debug message as local, we can declare and initialize this variable outside the loop

**Review by fmherschel — CHANGES_REQUESTED** [review]
See my comments

**Comment by angelabriel** [comment]
new commit to fix the discussed problems

**Comment by angelabriel** [comment]
Hopefully the problem bit hdbnsutil output is fixed in both situations

**Review by fmherschel — APPROVED** [review]
Looks good for me, thanks. We should test the new code base.

**Pull request merged**